### PR TITLE
[Fix](partial update) Fix `__DORIS_SEQUENCE_COL__` is not set for newly inserted rows in partial update

### PR DESCRIPTION
**Pull request by bobhan1**
### 1. Fix `__DORIS_SEQUENCE_COL__` is not set for newly inserted rows in partial update
before:
```sql
MySQL root@127.1:d1> CREATE TABLE IF NOT EXISTS t3 (
                  ->   `k` BIGINT NOT NULL,
                  ->   `c1` int,
                  ->   `c2` datetime default current_timestamp,
                  ->  ) UNIQUE KEY(`k`)
                  ->  DISTRIBUTED BY HASH(`k`) BUCKETS 1
                  ->  PROPERTIES (
                  ->  "replication_allocation" = "tag.location.default: 1",
                  ->  "enable_unique_key_merge_on_write" = "true",
                  ->  "function_column.sequence_col" = "c2"
                  ->  );
Query OK, 0 rows affected
MySQL root@127.1:d1> set enable_insert_strict=false;
MySQL root@127.1:d1> set enable_unique_key_partial_update=true;
MySQL root@127.1:d1> insert into t3(k,c1) values(99,99);
MySQL root@127.1:d1> set show_hidden_columns=true;
MySQL root@127.1:d1> select * from t3;
+----+----+---------------------+-----------------------+-----------------------+------------------------+
| k  | c1 | c2                  | __DORIS_DELETE_SIGN__ | __DORIS_VERSION_COL__ | __DORIS_SEQUENCE_COL__ |
+----+----+---------------------+-----------------------+-----------------------+------------------------+
| 99 | 99 | 2024-09-02 11:03:09 | 0                     | 2                     | <null>                 |
+----+----+---------------------+-----------------------+-----------------------+------------------------+
```
after:
```sql
MySQL root@127.1:d1> CREATE TABLE IF NOT EXISTS t3 (
                  ->   `k` BIGINT NOT NULL,
                  ->   `c1` int,
                  ->   `c2` datetime default current_timestamp,
                  ->  ) UNIQUE KEY(`k`)
                  ->  DISTRIBUTED BY HASH(`k`) BUCKETS 1
                  ->  PROPERTIES (
                  ->  "replication_allocation" = "tag.location.default: 1",
                  ->  "enable_unique_key_merge_on_write" = "true",
                  ->  "function_column.sequence_col" = "c2"
                  ->  );
Query OK, 0 rows affected
MySQL root@127.1:d1> set enable_insert_strict=false;
MySQL root@127.1:d1> set enable_unique_key_partial_update=true;
MySQL root@127.1:d1> insert into t3(k,c1) values(1,10);
MySQL root@127.1:d1> set show_hidden_columns=true;
MySQL root@127.1:d1> select * from t3;
+---+----+---------------------+-----------------------+-----------------------+------------------------+
| k | c1 | c2                  | __DORIS_DELETE_SIGN__ | __DORIS_VERSION_COL__ | __DORIS_SEQUENCE_COL__ |
+---+----+---------------------+-----------------------+-----------------------+------------------------+
| 1 | 10 | 2024-09-02 16:49:50 | 0                     | 2                     | 2024-09-02 16:49:50    |
+---+----+---------------------+-----------------------+-----------------------+------------------------+
```
### 2. Fix `current_timestamp()` precision loss for newly inserted rows in partial update
before:
```sql
MySQL root@127.1:d1> CREATE TABLE IF NOT EXISTS t3 (
                  ->   `k` BIGINT NOT NULL,
                  ->   `c1` int,
                  ->   `c2` datetime(6) default current_timestamp(6),
                  ->  ) UNIQUE KEY(`k`)
                  ->  DISTRIBUTED BY HASH(`k`) BUCKETS 1
                  ->  PROPERTIES (
                  ->  "replication_allocation" = "tag.location.default: 1",
                  ->  "enable_unique_key_merge_on_write" = "true",
                  ->  "function_column.sequence_col" = "c2"
                  ->  );
Query OK, 0 rows affected
MySQL root@127.1:d1> set enable_unique_key_partial_update=true;
MySQL root@127.1:d1> set enable_insert_strict=false;
MySQL root@127.1:d1> insert into t3(k,c1) values(3,10);
MySQL root@127.1:d1> set show_hidden_columns=true;
MySQL root@127.1:d1> select * from t3;
+---+----+----------------------------+-----------------------+-----------------------+----------------------------+
| k | c1 | c2                         | __DORIS_DELETE_SIGN__ | __DORIS_VERSION_COL__ | __DORIS_SEQUENCE_COL__     |
+---+----+----------------------------+-----------------------+-----------------------+----------------------------+
| 3 | 10 | 2024-09-02 19:04:55        | 0                     | 2                     | <null>                     |
+---+----+----------------------------+-----------------------+-----------------------+----------------------------+
```
after:
```sql
MySQL root@127.1:d1> CREATE TABLE IF NOT EXISTS t3 (
                  ->   `k` BIGINT NOT NULL,
                  ->   `c1` int,
                  ->   `c2` datetime(6) default current_timestamp(6),
                  ->  ) UNIQUE KEY(`k`)
                  ->  DISTRIBUTED BY HASH(`k`) BUCKETS 1
                  ->  PROPERTIES (
                  ->  "replication_allocation" = "tag.location.default: 1",
                  ->  "enable_unique_key_merge_on_write" = "true",
                  ->  "function_column.sequence_col" = "c2"
                  ->  );
Query OK, 0 rows affected
MySQL root@127.1:d1> set enable_unique_key_partial_update=true;
MySQL root@127.1:d1> set enable_insert_strict=false;
MySQL root@127.1:d1> insert into t3(k,c1) values(3,10);
MySQL root@127.1:d1> set show_hidden_columns=true;
MySQL root@127.1:d1> select * from t3;
+---+----+----------------------------+-----------------------+-----------------------+----------------------------+
| k | c1 | c2                         | __DORIS_DELETE_SIGN__ | __DORIS_VERSION_COL__ | __DORIS_SEQUENCE_COL__     |
+---+----+----------------------------+-----------------------+-----------------------+----------------------------+
| 3 | 10 | 2024-09-02 19:04:55.464438 | 0                     | 2                     | 2024-09-02 19:04:55.464438 |
+---+----+----------------------------+-----------------------+-----------------------+----------------------------+
```

branch-2.1-pick: https://github.com/apache/doris/pull/40964
branch-2.0-pick: https://github.com/apache/doris/pull/40966

### CODE REVIEW

**Comment by bobhan1** [comment]
run buildall

**Comment by bobhan1** [comment]
run buildall

**Comment by bobhan1** [comment]
run buildall

**Comment by bobhan1** [comment]
run buildall

**Comment by bobhan1** [comment]
run buildall

**Comment by bobhan1** [comment]
run buildall

**Comment by bobhan1** [comment]
run buildall

**Comment by bobhan1** [comment]
run buildall

**Comment by zhannngchen** [comment]
current solution is tricky, we'd better to make the default value of sequence column save with the mapping column

**Comment by bobhan1** [comment]
run buildall

**Review by zhannngchen — DISMISSED** [review]
LGTM

File: regression-test/suites/unique_with_mow_p0/partial_update/test_partial_update_seq_map_col.groovy [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+                `c2` datetime(6) null default current_timestamp(6),
+                ) UNIQUE KEY(`k`)
+                DISTRIBUTED BY HASH(`k`) BUCKETS 1
+                PROPERTIES (
+                "replication_allocation" = "tag.location.default: 1",
+                "enable_unique_key_merge_on_write" = "true",
+                "function_column.sequence_col" = "c2",
+                "store_row_column" = "${use_row_store}"); """ 
+            sql "insert into ${tableName}(k,c1) values(1,1);"
+            sql "insert into ${tableName}(k,c1) values(2,2);"
```
Comment (zhannngchen) [inline regression-test/suites/unique_with_mow_p0/partial_update/test_partial_update_seq_map_col.groovy:47]: Add some update operation, all operations are currently insert operations

**Comment by bobhan1** [comment]
run buildall

**Review by zhannngchen — DISMISSED** [review]

**Review by zhannngchen — APPROVED** [review]
LGTM

**Comment by bobhan1** [comment]
run buildall

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**